L1Trigger/L1TTrackMatch/interface/L1TkMuCorrDynamicWindows.h

```diff
@@ -0,0 +1,150 @@
+
+  // ------------------------------
+  static std::vector<double> prepare_corr_bounds(const string& fname, const string& hname);
+
+  void set_safety_factor(float sf_l, float sf_h) {
+    safety_factor_l_ = sf_l;
+    safety_factor_h_ = sf_h;
+    // std::cout << "L1TkMuCorrDynamicWindows : safety factor LOW is " << safety_factor_l_ << std::endl;
+    // std::cout << "L1TkMuCorrDynamicWindows : safety factor HIGH is " << safety_factor_h_ << std::endl;
+  }
+  void set_sf_initialrelax(float sf_l, float sf_h) {
+    initial_sf_l_ = sf_l;
+    initial_sf_h_ = sf_h;
+    // std::cout << "L1TkMuCorrDynamicWindows : initial relax safety factor LOW is " << initial_sf_l_ << std::endl;
+    // std::cout << "L1TkMuCorrDynamicWindows : initial relax safety factor HIGH is " << initial_sf_h_ << std::endl;
+  }
+  void set_relaxation_pattern(float pt_start, float pt_end) {
+    pt_start_ = pt_start;
+    pt_end_ = pt_end;
+    // std::cout << "L1TkMuCorrDynamicWindows : set relaxing from " << pt_start_ << " to " << pt_end_ << std::endl;
+  }
+  void set_safety_factor(float sf) { set_safety_factor(sf, sf); }
+  void set_sf_initialrelax(float sf) { set_sf_initialrelax(sf, sf); }
+  void set_do_relax_factor(bool val) {
+    do_relax_factor_ = val;
+    // std::cout << "L1TkMuCorrDynamicWindows : set do_relax to " << std::boolalpha << do_relax_factor_ << std::noboolalpha << std::endl;
+  }
+
+  void set_do_trk_qual_presel(bool val) { track_qual_presel_ = val; }
+
+  // setters for trk
+  void set_n_trk_par(int val) { nTrkPars_ = val; }
+  void set_min_trk_p(float val) { min_trk_p_ = val; }
+  void set_max_trk_aeta(float val) { max_trk_aeta_ = val; }
+  void set_max_trk_chi2(float val) { max_trk_chi2_ = val; }
+  void set_min_trk_nstubs(int val) { min_trk_nstubs_ = val; }
+
+  // getters for trk
+  const int n_trk_par() { return nTrkPars_; }
+  const float min_trk_p() { return min_trk_p_; }
+  const float max_trk_aeta() { return max_trk_aeta_; }
+  const float max_trk_chi2() { return max_trk_chi2_; }
+  const int min_trk_nstubs() { return min_trk_nstubs_; }
+
+private:
+  int findBin(double val);
+
+  // resolves ambiguities to give max 1 tkmu per EMTF
+  // if a pointer to narbitrated is passed, this vector is filled with the number of tracks arbitrated that were matched to the same EMTF
+  std::vector<int> make_unique_coll(const unsigned int& l1musSize,
+                                    const L1TTTrackCollectionType& l1trks,
+                                    const std::vector<int>& matches);
+
+  // converters
+  double deg_to_rad(double x) { return (x * angle_units::degPerRad); }
+
+  double eta_to_theta(double x) {
+    //  give theta in rad
+    return (2. * atan(exp(-1. * x)));
+  }
+
+  double to_mpio2_pio2(double x) {
+    //  put the angle in radians between -pi/2 and pi/2
+    while (x >= 0.5 * M_PI)
+      x -= M_PI;
+    while (x < -0.5 * M_PI)
+      x += M_PI;
+    return x;
+  }
+
+  double to_mpi_pi(double x) {
```

Comment:
prefer to stay with here defined function to avoid possible introduction of bugs.   can be done in subsequent PRs.  
There is still more code for this package to be integrated and being developed.